tests/src/test_cases/test_btnmatrix.c

```diff
@@ -0,0 +1,390 @@
+    lv_btnmatrix_set_map(btnm, btn_map);
+
+    lv_btnmatrix_set_btn_ctrl_all(btnm, LV_BTNMATRIX_CTRL_CHECKABLE);
+    lv_btnmatrix_set_one_checked(btnm, true);
+
+    lv_btnmatrix_set_btn_ctrl(btnm, 0, LV_BTNMATRIX_CTRL_CHECKED);
+    lv_btnmatrix_set_btn_ctrl(btnm, 1, LV_BTNMATRIX_CTRL_CHECKED);
+    lv_btnmatrix_set_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKED);
+    lv_btnmatrix_set_btn_ctrl(btnm, 3, LV_BTNMATRIX_CTRL_CHECKED);
+
+    TEST_ASSERT_TRUE(lv_btnmatrix_get_one_checked(btnm));
+
+    TEST_ASSERT_FALSE(lv_btnmatrix_has_btn_ctrl(btnm, 0, LV_BTNMATRIX_CTRL_CHECKED));
+    TEST_ASSERT_FALSE(lv_btnmatrix_has_btn_ctrl(btnm, 1, LV_BTNMATRIX_CTRL_CHECKED));
+    TEST_ASSERT_FALSE(lv_btnmatrix_has_btn_ctrl(btnm, 2, LV_BTNMATRIX_CTRL_CHECKED));
+    TEST_ASSERT_TRUE(lv_btnmatrix_has_btn_ctrl(btnm, 3, LV_BTNMATRIX_CTRL_CHECKED));
+}
+
+void test_btn_matrix_get_btn_text_works(void)
+{
+    static const char * btn_map[] = {"A", "B", "\n", "C", "D", ""};
+    lv_btnmatrix_set_map(btnm, btn_map);
+
+    const char * lv_btnmatrix_get_btn_text(const lv_obj_t * obj, uint16_t btn_id);
+
+    TEST_ASSERT_EQUAL_STRING("A", lv_btnmatrix_get_btn_text(btnm, 0));
+    TEST_ASSERT_EQUAL_STRING("B", lv_btnmatrix_get_btn_text(btnm, 1));
+    TEST_ASSERT_EQUAL_STRING("C", lv_btnmatrix_get_btn_text(btnm, 2));
+    TEST_ASSERT_EQUAL_STRING("D", lv_btnmatrix_get_btn_text(btnm, 3));
+}
+
+/* Common event handler for all the consecutive test cases. */
+static void event_handler(lv_event_t * e)
+{
+    lv_event_code_t code = lv_event_get_code(e);
+    if(code == exp_evt_code) {
+        event_triggered = true;
+    }
+}
+
+void test_btn_matrix_pressed_event_works(void)
+{
+    static const char * btn_map[] = {"A", "B", "\n", "C", "D", ""};
+    lv_btnmatrix_set_map(btnm, btn_map);
+    lv_obj_add_event_cb(btnm, event_handler, LV_EVENT_PRESSED, NULL);
+    /* Set expected event code before the event is raised. */
+    exp_evt_code = LV_EVENT_PRESSED;
+    /* Click button index 0. */
+    lv_test_mouse_click_at(10, 10);
+    TEST_ASSERT_TRUE(event_triggered);
+}
+
+void test_btn_matrix_release_event_works(void)
+{
+    static const char * btn_map[] = {"A", "B", "\n", "C", "D", ""};
+    lv_btnmatrix_set_map(btnm, btn_map);
+    lv_btnmatrix_set_btn_ctrl_all(btnm, LV_BTNMATRIX_CTRL_CHECKABLE);
+    lv_obj_add_event_cb(btnm, event_handler, LV_EVENT_RELEASED, NULL);
+    /* Set expected event code before the event is raised. */
+    exp_evt_code = LV_EVENT_RELEASED;
+
+    /* Click button index 0. */
+    lv_test_mouse_click_at(65, 35);
+    TEST_ASSERT_TRUE(event_triggered);
+    event_triggered = false;
+    /* This will increase test coverage by unchecking the
+     same button. */
+    lv_btnmatrix_set_btn_ctrl(btnm, 0, LV_BTNMATRIX_CTRL_POPOVER);
+    /* Click button index 0. */
+    lv_test_mouse_click_at(65, 35);
+    TEST_ASSERT_TRUE(event_triggered);
+}
+
+void test_btn_matrix_size_changed_event_works(void)
```

Comment:
It's not required because it test only that LVGL really sends a `LV_EVENT_SIZE_CHANGED` when you send a `LV_EVENT_SIZE_CHANGED` event manually. It's not specific to the button matrix, but to the events in general.